apps/etterlatte-behandling/src/main/kotlin/Tilgangsstyring.kt

```diff
@@ -43,29 +46,60 @@ val adressebeskyttelsePlugin: RouteScopedPlugin<PluginConfiguration> = createRou
 ) {
     on(AdressebeskyttelseHook) { call ->
         val bruker = call.bruker
+
         if (bruker is SystemBruker) {
             return@on
         }
         if (call.request.uri.contains(TILGANG_ROUTE_PATH)) {
             return@on
         }
-        val behandlingId = call.parameters[BEHANDLINGSID_CALL_PARAMETER]
 
-        if (!behandlingId.isNullOrEmpty()) {
-            if (pluginConfig.behandlingIdHarAdressebeskyttelse(behandlingId)) {
-                call.respond(HttpStatusCode.NotFound)
+        if (bruker is Saksbehandler) {
+            val behandlingId = call.parameters[BEHANDLINGSID_CALL_PARAMETER]
+            if (!behandlingId.isNullOrEmpty()) {
+                if (!pluginConfig.harTilgangBehandling(behandlingId, SaksbehandlerMedRoller(bruker))) {
+                    call.respond(HttpStatusCode.NotFound)
+                }
+                return@on
             }
-            return@on
-        }
 
-        val sakId = call.parameters[SAKID_CALL_PARAMETER]
-        if (!sakId.isNullOrEmpty()) {
-            if (pluginConfig.sakIdHarAdressebeskyttelse(sakId)) {
-                call.respond(HttpStatusCode.NotFound)
+            val sakId = call.parameters[SAKID_CALL_PARAMETER]
+            if (!sakId.isNullOrEmpty()) {
+                if (!pluginConfig.harTilgangTilSak(sakId.toLong(), SaksbehandlerMedRoller(bruker))) {
+                    call.respond(HttpStatusCode.NotFound)
+                }
+                return@on
             }
-            return@on
         }
 
         return@on
     }
+}
+
+data class SaksbehandlerMedRoller(val saksbehandler: Saksbehandler) {
```

Comment:
Ref diskusjonen over, så tror jeg vi bør bli enige om hvilke som benyttes. Hvis vi skal benytte den som ligger på context, så kan vel egentlig alt dette + wiring i tjenestene fjernes, og gjøre noe slik i tjenestelaget:

```kotlin
    override fun harTilgangTilSak(sakId: Long, saksbehandlerMedRoller: SaksbehandlerMedRoller): Boolean {
        val saksbehandler = Kontekst.get().AppUser as Saksbehandler

        val sakMedGradering = dao.hentSakMedGradering(sakId) ?: return true
        return when (sakMedGradering.adressebeskyttelseGradering) {
            AdressebeskyttelseGradering.STRENGT_FORTROLIG_UTLAND -> saksbehandler.harRolleStrengtFortrolig()
            AdressebeskyttelseGradering.STRENGT_FORTROLIG -> saksbehandler.harRolleStrengtFortrolig()
            AdressebeskyttelseGradering.FORTROLIG -> saksbehandler.harRolleFortrolig()
            AdressebeskyttelseGradering.UGRADERT -> true
            else -> true
        }
    }
```
Føles litt merkelig å ha denne logikken to steder 🤔

Reply:
Du vil ha det her?

Reply:
Ikke nødvendigvis, men det virker litt tungvindt/dobbelt opp å gjøre alt dette når vi allerede har informasjonen vi trenger i context 😅 Det ryddigste er sikkert å fjerne AppUser fra context, og benytte bruker slik vi gjør de fleste andre steder? Litt skeptisk til å innføre en tredje variant.

Reply:
`val saksbehandler = Kontekst.get().AppUser as Saksbehandler ` er fjernet, og målet er å kun bruke det som ligger på `PipelineContext<*, ApplicationCall>` men vil ikke røre det i denne PRen.

Reply:
https://jira.adeo.no/browse/EY-2013